src/librustc/hir/lowering.rs

```diff
@@ -990,15 +990,14 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         }
     }
 
-    fn lower_attrs_extendable(&mut self, attrs: &[Attribute]) -> Vec<Attribute> {
-        attrs
-            .iter()
-            .map(|a| self.lower_attr(a))
-            .collect()
+    fn lower_attrs_extendable(&mut self, attrs: &[Attribute]) -> &'hir [Attribute] {
```

Comment:
This should still return `Vec<Attribute>` since callers may want to extend it.

Reply:
`lower_attrs` is the function which should return `&'hir [Attribute]`.

Reply:
I suggest you add a `lower_attrs_arena` variant which returns `&'hir [Attribute]` which can be removed when all the attributes are migrated over.